core/src/main/java/ai/timefold/solver/core/impl/solver/scope/SolverScope.java

```diff
@@ -346,6 +356,7 @@ public SolverScope<Solution_> createChildThreadSolverScope(ChildThreadType child
         // Experiments show that this trick to attain reproducibility doesn't break uniform distribution
         childThreadSolverScope.workingRandom = new Random(workingRandom.nextLong());
         childThreadSolverScope.scoreDirector = scoreDirector.createChildThreadScoreDirector(childThreadType);
+        childThreadSolverScope.setValueRangeResolver(scoreDirector.getValueRangeResolver());
```

Comment:
Child thread has its own score director, therefore it needs to have its own value range resolver as well. Score director will be cloning solutions.